### PR TITLE
Updating gaiad to version 4.1.0 and updating proto-compiler to use ibc-go

### PR DESCRIPTION
**Pull request by andynog**
Closes: #702 

## Description

Updated gaiad to the latest version 4.1.0 (released today March 10, 2021 which uses cosmos-sdk 0.42 but includes a security fix so all nodes upgraded to this version)
______

For contributor use:

- [x] Updated the __Unreleased__ section of [CHANGELOG.md](https://github.com/informalsystems/ibc-rs/blob/master/CHANGELOG.md) with the issue.
- [ ] If applicable: Unit tests written, added test to CI.
- [X] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Updated relevant documentation (`docs/`) and code comments.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

**Comment by andynog** [comment]
I think this is good to be merged @adizere and @ancazamfir.

**Comment by ancazamfir** [comment]
Thanks! We also need to fix the scripts, `./scripts/dev-env ~/.hermes/config.toml ibc-0 ibc-1` currently fails. I realized that running `./e2e/run.py  -c ~/.hermes/config.toml` might be ok once the scripts are fixed as we currently don't decode in `master` the new `header` field in the update client result (this causes failures in my PR #691 where I do this)

**Comment by ancazamfir** [comment]
We should also update the protos, now in https://github.com/cosmos/ibc-go/tree/main/proto/ibcgo

**Comment by andynog** [comment]
> Thanks! We also need to fix the scripts, `./scripts/dev-env ~/.hermes/config.toml ibc-0 ibc-1` currently fails. I realized that running `./e2e/run.py -c ~/.hermes/config.toml` might be ok once the scripts are fixed as we currently don't decode in `master` the new `header` field in the update client result (this causes failures in my PR #691 where I do this)

I've tested the `dev-env` script with `gaiad v4.1.0` and it wokrs for me.

**Comment by andynog** [comment]
Got logic to compile the protos from `ibc-go` for ibc protos and from `cosmos-sdk` for `cosmos` protos working. 

I could compile all crates with the new protos. Ran a `dev-env` test and it passed.

I think this is good to go

File: proto/src/prost/tendermint.abci.rs [hunk 1]
```diff
@@ -0,0 +1 @@
...lt () ; let path = http :: uri :: PathAndQuery :: from_static ("/tendermint.abci.ABCIApplication/LoadSnapshotChunk") ; self . inner . unary (request . into_request () , path , codec) . await } pub async fn apply_snapshot_chunk (& mut self , request : impl tonic :: IntoRequest < :: tendermint_proto :: abci :: RequestApplySnapshotChunk > ,) -> Result < tonic :: Response < :: tendermint_proto :: abci :: ResponseApplySnapshotChunk > , tonic :: Status > { self . inner . ready () . await . map_err (| e | { tonic :: Status :: new (tonic :: Code :: Unknown , format ! ("Service was not ready: {}" , e . into ())) }) ? ; let codec = tonic :: codec :: ProstCodec :: default () ; let path = http :: uri :: PathAndQuery :: from_static ("/tendermint.abci.ABCIApplication/ApplySnapshotChunk") ; self . inner . unary (request . into_request () , path , codec) . await } } impl < T : Clone > Clone for AbciApplicationClient < T > { fn clone (& self) -> Self { Self { inner : self . inner . clone () , } } } impl < T > std :: fmt :: Debug for AbciApplicationClient < T > { fn fmt (& self , f : & mut std :: fmt :: Formatter < '_ >) -> std :: fmt :: Result { write ! (f , "AbciApplicationClient {{ ... }}") } } }
```
Comment (ancazamfir) [inline proto/src/prost/tendermint.abci.rs:1]: Is there a way to not generate these `proto/src/prost/tendermint*` empty files?

**Review by ancazamfir — APPROVED** [review]

File: proto/src/prost/tendermint.abci.rs (hunk 1 above)
Comment (adizere) [reply proto/src/prost/tendermint.abci.rs:1]: It seems that the empty files related to tendermint are dependencies in other files, and that's why they get generated. I manually removed the empty ones from this commit. I left the abci.rs.

**Pull request merged**